### DISCUSSION TITLE
How can i do a text color wipe animation in bevy ? (gif attached)

### DISCUSSION THREAD

**Discussion by neevany** [bevyengine/bevy#12975] [Q&A]
![how-can-i-do-a-text-color-wipe-animation-in-bevy-v0-m7vepq5rjkuc1](https://github.com/bevyengine/bevy/assets/18303579/001773a1-258b-4662-be2b-85141383f982)

**Comment by torsteingrindvik** [comment]
One way that should work is a two pass approach:

1. [render_ui_to_texture](https://github.com/bevyengine/bevy/blob/main/examples/ui/render_ui_to_texture.rs) which would allow you to render the text in the first color to a texture
2. [shader_material_2d](https://github.com/bevyengine/bevy/blob/main/examples/shader/shader_material_2d.rs) which could load the result from the first step, then apply the wipe effect in the pixel shader if you give the shader enough data about the wipe progress and the proximity to the cursor

**Reply by torsteingrindvik** [reply]
Might be possible in a single pass via [ui_material](https://github.com/bevyengine/bevy/blob/main/examples/ui/ui_material.rs)

**Reply by PiergiorgioZagaria** [reply]
I was trying to do something similar, adding a rainbow colored or wavy text effect to use it with bevy yarnsnipper markup.

The problem with your first approach, in my use case, is that you need to find the right size of the texture by trial and error (I don't know if there is a way to know how much space a phrase can take given the font and the font size), also you need to make sure that the background matches.

Could you explain the single pass approach? I read the example, and from what I understood you could make a textbundle and then a materialbundle as a child of the textbundle, to render the material on top of the text. But I don't know how to get the color of the underlying text in the shader. Maybe this approach is completely off though
